gladier_xpcs/tools/gather_xpcs_metadata.py

```diff
@@ -152,15 +152,31 @@ def gather(dataframe):
         # Parent: sanat
         metadata['parent'] = re.search(r'([a-z]+)*', root_folder.name).group()
     except Exception:
-        p = pathlib.Path(event['proc_dir']) / 'gather_xpcs_metadata.error'
+        p = pathlib.Path(data['proc_dir']) / 'gather_xpcs_metadata.error'
         with open(str(p), 'w+') as f:
             f.write(traceback.format_exc())
 
-
-    pilot = event['pilot']
+    try:
+        import boost_corr
+        boost_version = boost_corr.__version__
+        gpu = data.get('gpu_flag','0')
+    except:
+        boost_version = 0
+        gpu = None
+
+    pilot = data['pilot']
     # metadata passed through from the top level takes precedence. This allows for
     # overriding fields through $.input
     metadata.update(pilot.get('metadata', {}))
+    metadata.update({
+        'executable' : {
+            'name': 'boost_corr',
+            'version': boost_version,
+            'device': gpu,
+            'source': 'https://pypi.org/project/boost_corr/',
+            }
+        })
+
```

Comment:
It would be good to put this line after the executable update line: 
```
metadata.update(pilot.get('metadata', {}))
```
That way it can be overrided by input if the `gather` step can't determine the tooling for some reason.